route/matcher.go

```diff
@@ -23,3 +24,10 @@ func prefixMatcher(uri string, r *Route) bool {
 func globMatcher(uri string, r *Route) bool {
 	return r.Glob.Match(uri)
 }
+
+// noCase matches path to the routes' path ignoring case
+func noCaseMatcher(uri string, r *Route) bool {
+	lowerURI := strings.ToLower(uri)
+	lowerPath := strings.ToLower(r.Path)
+	return strings.HasPrefix(lowerURI, lowerPath)
```

Comment:
After looking at this again, I think this should be a case insensitive _prefix_ match. Originally I was using `EqualFold`. Thoughts?

Reply:
```go
// todo(fs): if this turns out to be a performance issue we should cache
// todo(fs): strings.ToLower(r.Path) in r.PathLower
return strings.HasPrefix(strings.ToLower(uri), strings.ToLower(r.Path))
```